source-code-git/fs/src/interface.ts

```diff
@@ -15,9 +15,7 @@ export type NodeishFilesystem = {
 	 *
 	 * Upon success, fulfills with undefined if recursive is false, or the first directory path created if recursive is true.
 	 */
-	mkdir:
-		| ((path: string) => Promise<void>)
-		| ((path: string, options?: { recursive: boolean }) => Promise<string>)
+	mkdir: (path: string, options?: { recursive: boolean }) => Promise<string | undefined>
```

Comment:
That is incorrect. The API I defined was correct. Node returns Promise<void> if no options are provided and a string if options are provided. 

See https://nodejs.org/api/fs.html#fspromisesmkdirpath-options. 

I will revert this change because we narrow the types, leading to better DX and consistency.

Reply:
Tada. Narrowing the type immediately revealed a buy in memoryFs

<img width="1185" alt="CleanShot 2023-04-28 at 11 08 15@2x" src="https://user-images.githubusercontent.com/35429197/235106037-f8ba9081-1caa-474a-948e-1d7caba0dd8a.png">

Reply:
NVM. 

- your implementation had no error. typescript was not widening "not implemented" to a string 

- TypeScript is not smart enough to match overloads to a naive type as node:fs/promises uses then :(

<img width="813" alt="CleanShot 2023-04-28 at 11 58 53@2x" src="https://user-images.githubusercontent.com/35429197/235117803-99215bbe-6370-4901-aa1c-22c4ae1c1873.png">


Okay, no overloading and narrow types then (at least for now. Maybe TypeScript wizard @ivanhofer knows how to fix this in the future).